robotoff/insights/importer.py

```diff
@@ -801,23 +729,32 @@ def import_insights(
     if product_store is None:
         product_store = get_product_store()
 
-    importers: Dict[InsightType, BaseInsightImporter] = {}
-    product_predictions = sorted(product_predictions, key=operator.attrgetter("type"))
+    importers: Dict[InsightType, Optional[InsightImporter]] = {}
+    product_predictions = [
+        p for p in product_predictions if is_valid_product_predictions(p, product_store)
+    ]
+    predictions_imported = import_product_predictions(
+        product_predictions, server_domain
+    )
+    logger.info(f"{predictions_imported} predictions imported")
 
     prediction_type: PredictionType
     insight_group: Iterable[ProductPredictions]
     imported = 0
 
     for prediction_type, insight_group in itertools.groupby(
-        product_predictions, operator.attrgetter("type")
+        sorted(product_predictions, key=operator.attrgetter("type")),
+        operator.attrgetter("type"),
     ):
         insight_type = InsightType[prediction_type]
         if insight_type not in importers:
             importers[insight_type] = InsightImporterFactory.create(
                 insight_type, product_store
             )
-
         importer = importers[insight_type]
-        imported += importer.import_insights(insight_group, server_domain, automatic)
+        if importer is not None:
+            imported += importer.import_insights(
+                insight_group, server_domain, automatic
+            )
```

Comment:
Isn't it the case to log a warning if there is no importer ? (or is it supposed to be normal ?)

Reply:
Some prediction (such as `image_orientation`) shouldn't be imported as insights, so it's expected that some importers are missing for some prediction types.
The future way of dealing with insight generation will be a bit different, as each insight type will register the prediction types that are necessary to generate the insights. If new predictions of one of these prediction types were imported, the insight generation for the associated insight type will be triggered.